include/irr/asset/format/encodePixels.h

```diff
@@ -30,18 +30,18 @@ namespace irr { namespace video
 			return -1; // handle an error
 	}
 
-	template<typename T, typename U>
-	inline void clampVariableProperly(T& variableToAssignClampingTo, const U& variableToClamp, bool isItNormalizedVariable = true)
+	template<typename T>
+	inline void clampVariableProperly(T& variableToAssignClampingTo, const double& variableToClamp, const asset::E_FORMAT &format)
```

Comment:
make format a template parameter.

Reply:
Is it a good practice or it goes on performance or you want it to be as you said just simply?

Reply:
template parameter basically is guaranteed inlining and duplicate code generation for every value of `format`, a parameter the compiler may still leave some code to deal with a mutable `format`